wordpress-mobile/WordPress-Android
# [Reader] Fix tag posts list paging

Fixes #19879 

We were sorting the tag related blog posts comparing `date_published` as a string instead of a date. This was causing paging not to work properly.

I've replaced `...ORDER BY date_published...` with `...ORDER BY datetime(date_published)...` in our query so the posts are sorted as dates correctly.

### Before
```
SELECT <columns> FROM tbl_posts WHERE tag_name=? AND tag_type=? ORDER BY date_published DESC
```

https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/a007afc1-801d-4f21-87ab-e74a9f6130ce

### After
```
SELECT <columns> FROM tbl_posts WHERE tag_name=? AND tag_type=? ORDER BY datetime(date_published) DESC
```

https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/ec7a0768-f2ad-4768-8b87-1418e9775972

-----

## To Test:
Please test this change with different SQLite versions. Also compare the results with web to make sure now we're displaying the same results using the same order.

- Install JP and sign-in
- Open "Reader"
- Select a tag
- Start scrolling: it will stop loading new posts at some point, even though there are many more (compare with web)

Another way to test is opening the tag posts list from blogging prompts card:
- Install JP and sign-in
- Open "My Site"
- Tap on "View all responses" in blogging prompts card
- Start scrolling: it will stop loading new posts at some point, even though there are many more (compare with web)

-----

## Regression Notes

1. Potential unintended areas of impact

    - SQLite function `datetime(...)` might not available in every version?

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    --

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
